core/test-api/db/src/main/java/org/opennms/core/test/db/TemporaryDatabasePostgreSQL.java

```diff
@@ -117,19 +129,35 @@ public TemporaryDatabasePostgreSQL(String testDatabase, String driver, String ur
     public TemporaryDatabasePostgreSQL(String testDatabase, String driver, String url,
                              String adminUser, String adminPassword, boolean useExisting) throws Exception {
         // Append the current object's hashcode to make this value truly unique
-        m_testDatabase = testDatabase;
+        m_testDatabase = testDatabase != null ? testDatabase : getDatabaseName(this);
         m_driver = driver;
         m_url = url;
         m_adminUser = adminUser;
         m_adminPassword = adminPassword;
         m_useExisting = useExisting;
     }
 
+    protected static void failIfUnitTest() throws TemporaryDatabaseException {
+        if ("true".equals(System.getProperty(OPENNMS_UNIT_TEST_PROPERTY))) {
+            throw new TemporaryDatabaseException("The '" + OPENNMS_UNIT_TEST_PROPERTY + "' property is set to true, " +
+                    "however this class is only suitable for integration tests, not unit tests. " +
+                    "Please refactor to not use a database or move to an integration test (name your test class *IT.java). " +
+                    "See http://wiki.opennms.org/wiki/Test_conventions for details");
+        }
+    }
+
+    protected static String getDatabaseName(Object hashMe) {
+        // Append the current object's hashcode to make this value truly unique
+        return String.format("opennms_test_%s_%06d_%s", System.currentTimeMillis(), System.nanoTime(), Math.abs(hashMe.hashCode()));
```

Comment:
System.currentTimeMillis() provides a wall clock reference for coordinating with external entities (test logs, etc.) and System.nanoTime() to ensure uniqueness. I could move the System.currentTimeMillis() to the blame table.